### ISSUE TITLE
Use best performing model for test loop

### ISSUE THREAD

**Issue by raphaelreinauer** [giotto-ai/giotto-deep#42]
In the current version, the model at the end of training is used to get the test accuracy.
https://github.com/giotto-ai/giotto-deep/blob/2965f49a5653bc51790bc76c66e21f2e5ca3e65d/gdeep/pipeline/pipeline.py#L150

Using the model with the highest validation accuracy (or the low validation loss) during training would be better.
The weights of that model have to be stored temporarily.
